lib/ansible/modules/cloud/amazon/elasticache_parameter_group.py

```diff
@@ -0,0 +1,333 @@
+    """ Modify ElastiCache parameter group to reflect the new information if it differs from the current. """
+    # compares current group parameters with the parameters we've specified to to a value to see if this will change the group
+    format_parameters = []
+    for key in values:
+        value = text_type(values[key])
+        format_parameters.append({'ParameterName': key, 'ParameterValue': value})
+    try:
+        response = conn.modify_cache_parameter_group(CacheParameterGroupName=name, ParameterNameValues=format_parameters)
+    except boto.exception.BotoServerError as e:
+        module.fail_json(msg="Unable to modify cache parameter group.", exception=traceback.format_exc(e))
+    return response
+
+def reset(module, conn, name, values):
+    """ Reset ElastiCache parameter group if the current information is different from the new information. """
+    # used to compare with the reset parameters' dict to see if there have been changes
+    old_parameters_dict = make_current_modifiable_param_dict(module, conn, name)
+
+    format_parameters = []
+
+    # determine whether to reset all or specific parameters
+    if values:
+        all_parameters = False
+        format_parameters = []
+        for key in values:
+            value = text_type(values[key])
+            format_parameters.append({'ParameterName': key, 'ParameterValue': value})
+    else:
+        all_parameters = True
+
+    try:
+        response = conn.reset_cache_parameter_group(CacheParameterGroupName=name, ParameterNameValues=format_parameters, ResetAllParameters=all_parameters)
+    except boto.exception.BotoServerError as e:
+        module.fail_json(msg="Unable to reset cache parameter group.", exception=traceback.format_exc(e))
+
+    # determine changed
+    new_parameters_dict = make_current_modifiable_param_dict(module, conn, name)
+    changed = check_changed_parameter_values(values, old_parameters_dict, new_parameters_dict)
+
+    return response, changed
+
+def get_info(conn, name):
+    """ Gets info about the ElastiCache parameter group. Returns false if it doesn't exist or we don't have access. """
+    try:
+        data = conn.describe_cache_parameters(CacheParameterGroupName=name)
+        return data
+    except botocore.exceptions.ClientError as e:
+        return False
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+        group_family    = dict(required=False, type='str', choices=['memcached1.4', 'redis2.6', 'redis2.8', 'redis3.2']),
+        name            = dict(required=True, type='str'),
+        description     = dict(required=False),
+        state           = dict(required=True),
+        values          = dict(required=False, type='dict'),
+    )
+    )
+    module = AnsibleModule(argument_spec=argument_spec)
+
+    if not HAS_BOTO3:
+        module.fail_json(msg='boto required for this module')
+
+    parameter_group_family  = module.params.get('group_family')
```

Comment:
Don't align things like this (it violates pep8 and is annoying if a longer thing gets added later)